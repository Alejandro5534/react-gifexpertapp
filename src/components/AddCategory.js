import { useState } from 'react'
import PropTypes from 'prop-types';
export const AddCategory = ({setCategories}) => {
    
    const [inputValue, setinputValue] = useState('')
    const handleInputChange = (e)=> {
        setinputValue(e.target.value);
    }
    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length>2) {
            setCategories(categories => [ inputValue,...categories])
            setinputValue('')
        }
        // console.log('Submit Hecho')
    }
    return (
          
            <form onSubmit = {handleSubmit}>    
                <input 
                    placeholder= "Escribe una o mas palabras clave para que encuentres, entre 10 GIFS, tu preferido"
                    type="text"
                    value = {inputValue}
                    onChange={handleInputChange}

                />
            </form>    
        
    )
}

AddCategory.prototype={
    setCategories: PropTypes.func.isRequired        
}