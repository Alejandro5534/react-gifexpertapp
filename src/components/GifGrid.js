// import { useState, useEffect } from "react";
// import { getGifs } from "../helpers/getGifs";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";

export const GifGrid = ({category}) => {
    // const [counter, setcounter] = useState(0);
    // const [images, setimages] = useState([])
    const {data:images,loading} = useFetchGifs(category);
    
    
    
    return(
        <>
        <h3 className="animate__animated animate__fadeIn"> {category}</h3>
        {loading && <p className="animate__animated animate__flash">Cargando..</p> }
        <div className="card-grid">
            
                {
                    images.map((img)=>(
                        <GifGridItem
                            key={img.id} 
                            {...img}
                        />    
                    ))
                }
            
            
        </div>
        </>
    );
}

