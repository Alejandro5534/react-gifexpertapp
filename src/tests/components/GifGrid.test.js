import {shallow} from 'enzyme'
// import {createSerializer} from 'enzyme-to-json'
// impourt { GifGrid } from '../../components/GifGrid';
import { GifGridItem } from '../../components/GifGridItem';

describe ('Pruebas a GifGrid',()=> {
    
    const title = 'Soy un titulo';
    const url = 'https://localhost/imagen.jpg';
    const wrapper = shallow(<GifGridItem title={title} url = {url}/>);
    test('debe de mostrar el componente correctamente', ()=>{
        expect(wrapper).toMatchSnapshot();
    })
    test('debe de tener un parrafo con el title', ()=>{
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title)
    })
    test('debe de tener la imagen igual al url y alt de los props', ()=>{
        const img = wrapper.find('img');
        // console.log(img.prop('src'))
        expect(img.prop('src')).toBe(url)
        expect(img.prop('alt')).toBe(title)
    })
    test('debe de tener animate_fadeIn', ()=>{
        const div = wrapper.find('div');
        const classNameRequired = 'animate__fadeIn'
        // console.log(div.prop('className'))
        expect(div.prop('className').includes(classNameRequired)).toBe(true)
    })

})