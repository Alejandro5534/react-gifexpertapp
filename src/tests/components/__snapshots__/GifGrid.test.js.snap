// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a GifGrid debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Soy un titulo"
    src="https://localhost/imagen.jpg"
  />
  <p>
    Soy un titulo
  </p>
</div>
`;
